crates/telio-starcast/src/transport.rs

```diff
@@ -228,6 +281,29 @@ impl Runtime for State {
     type Err = Error;
 
     async fn wait(&mut self) -> WaitResponse<'_, Self::Err> {
+        let transport_socket = match &self.transport_socket {
+            Some(transport_socket) => transport_socket.to_owned(),
+            None => {
+                PinnedSleep::new(self.exponential_backoff.get_backoff(), ()).await;
+                let Ok(transport_socket) = open_transport_port(
+                    self.socket_pool.clone(),
+                    self.meshnet_ip,
+                    MULTICAST_TRANSPORT_PORT,
+                )
+                .await
+                else {
+                    telio_log_warn!(
+                        "Starcast transport socket still not opened, will retry in {:?}",
+                        self.exponential_backoff.get_backoff()
+                    );
+                    self.exponential_backoff.next_backoff();
+                    return Self::next();
```

Comment:
What does it do, just runs `wait` function another time? If yes, I guess you could simplify it by just either waiting for sockets or waiting on sockets.

Reply:
I'd rather not block the wait loop. Especially, if external actions will need to be handled in the future.

Reply:
Hmm, maybe I'm missing sth, but it seems to me as the same amount of blocking it is currently, it would be just a bit simpler

Reply:
As I said, if in the future we needed to call `task_exec!` while we're still waiting for the transport socket to be opened, we'd have to revert your suggested changes, so I'd rather leave it as it is, since the simplicity here is kind of subjective.

Reply:
Ok